### PR TITLE
add 9 Inplace APIs that have core.ops implementation in its original APIs

### PR DESCRIPTION
**Pull request by pangyoki**
### PR types

New features

### PR changes

APIs

### Describe

Inplace Strategy has been implemented in PR #30103 .
Add 9 Inplace APIs in this PR.

|Inplace OP | fluid下API | paddle下API | 新增Inplace API |
|---|---|---|---|
| relu6 | paddle.fluid.layers.nn.relu6 | paddle.nn.functional.activation.relu6 |  paddle.nn.functional.activation.relu6_ | 
| hard_sigmoid | paddle.fluid.layers.nn.hardsigmoid | paddle.nn.functional.activation.hardsigmoid |  paddle.nn.functional.activation.hardsigmoid_ | 
| leaky_relu | paddle.fluid.layers.nn.leaky_relu | paddle.nn.functional.activation.leaky_relu | paddle.nn.functional.activation.leaky_relu_ | 
|  |  |  |  | 
| softmax_with_cross_entropy | paddle.fluid.layers.loss.softmax_with_cross_entropy | paddle.nn.functional.loss.softmax_with_cross_entropy |  paddle.nn.functional.loss.softmax_with_cross_entropy_ | 
|  |  |  |  | 
| sum | paddle.fluid.layers.nn.sum | paddle.tensor.math.add_n |  paddle.tensor.math.add_n_ | 
| clip | paddle.fluid.layers.nn.clip | paddle.tensor.math.clip | paddle.tensor.math.clip_  | 
| scale | paddle.fluid.layers.nn.scale | paddle.tensor.math.scale |  paddle.tensor.math.scale_ | 
| elementwise_sub | paddle.fluid.layers.nn.elementwise_sub | paddle.tensor.math.subtract |  paddle.tensor.math.subtract_ | 
| elementwise_add | paddle.fluid.layers.nn.elementwise_add | paddle.tensor.math.add | paddle.tensor.math.add_ |

### CODE REVIEW

File: python/paddle/fluid/tests/unittests/test_elementwise_sub_op.py [hunk 1]
```diff
@@ -289,6 +289,32 @@ def _executed_api(self):
...
+
+        inplace_result = paddle.subtract_(x, y)
+        numpy_result = x_numpy - y_numpy
+        self.assertEqual((inplace_result.numpy() == numpy_result).all(), True)
+        paddle.enable_static()
+
+    def test_broadcast_errors(self):
+        paddle.disable_static()
+        x = paddle.rand([3, 4])
+        y = paddle.rand([2, 3, 4])
```
Comment (zhiqiu) [inline python/paddle/fluid/tests/unittests/test_elementwise_sub_op.py]: add more comlecated broadcast testcases, e.g., [1,2,1,3] and [4,2,5,1]

File: python/paddle/tensor/math.py [hunk 2]
```diff
@@ -321,8 +321,12 @@ def add_(x, y, name=None):
     if in_dygraph_mode():
         op_type = 'elementwise_add_'
         axis = -1
-        return _elementwise_op_in_dygraph(
+        inplace_shape = x.shape
+        out = _elementwise_op_in_dygraph(
             x, y, axis=axis, op_name=op_type)
+        if inplace_shape != out.shape:
```
Comment (zhiqiu) [inline python/paddle/tensor/math.py]: The check should be down before the execution of the op.

File: python/paddle/fluid/tests/unittests/test_elementwise_sub_op.py (hunk 1 above)
Comment (pangyoki) [reply python/paddle/fluid/tests/unittests/test_elementwise_sub_op.py]: done

File: python/paddle/tensor/math.py (hunk 2 above)
Comment (pangyoki) [reply python/paddle/tensor/math.py]: done
